src/abstractWrappers/Disputes.js

```diff
@@ -648,100 +650,124 @@ class Disputes extends AbstractWrapper {
+      if (userData.appealCreatedAt) appealCreatedAt = userData.appealCreatedAt
+      if (userData.appealDeadlines) appealDeadlines = userData.appealDeadlines
+      if (userData.appealRuledAt) appealRuledAt = userData.appealRuledAt
+      if (userData.netPNK) netPNK = userData.netPNK
+      // eslint-disable-next-line no-unused-vars
+    } catch (err) {
+      // Fetching a dispute will fail if it hasn't been added to the store yet. This is ok, we can just not return store data
     }
 
-    // get evidence
-    const evidence = await this.getEvidenceForArbitrableContract(
-      arbitrableContractAddress
-    )
-
-    const firstSession = dispute.firstSession
+    // Build juror info and ruling arrays, indexed by appeal number
     const lastSession = dispute.firstSession + dispute.numberOfAppeals
-    // NOTE arrays indexed by appeal number
-    const appealRulings = []
     const appealJuror = []
-    for (let appeal = 0; appeal <= lastSession - firstSession; appeal++) {
-      // get ruling for appeal. Note appeal 0 is first session
-      const ruling = await this._Arbitrator.currentRulingForDispute(
-        arbitratorAddress,
-        disputeId,
-        appeal
-      )
-
-      appealRulings[appeal] = {
-        ruling,
-        voteCounter: dispute.voteCounters[appeal],
-        ruledAt: appealRuledAt[appeal],
-        deadline: appealDeadlines[appeal]
-      }
+    const appealRulings = []
+    for (let appeal = 0; appeal <= dispute.numberOfAppeals; appeal++) {
+      const isLastAppeal = dispute.firstSession + appeal === lastSession
 
-      const baseFee = dispute.arbitrationFeePerJuror
+      // Get appeal data
       const draws = appealDraws[appeal] || []
       let canRule = false
-      if (firstSession + appeal === lastSession && draws.length > 0) {
-        canRule = await this._Arbitrator.canRuleDispute(
+      let canRepartition = false
+      let canExecute = false
+      let ruling
+      const promises = [
+        this._Arbitrator.currentRulingForDispute(
           arbitratorAddress,
-          disputeId,
-          draws,
-          account
+          disputeID,
+          appeal
         )
+      ]
+
+      // Extra info for the last appeal
+      if (isLastAppeal) {
+        if (draws.length > 0)
+          promises.push(
+            this._Arbitrator.canRuleDispute(
+              arbitratorAddress,
+              disputeID,
+              draws,
+              account
+            )
+          )
+
+        if (session && period)
+          canRepartition =
+            lastSession <= session && // Not appealed to the next session
+            period === arbitratorConstants.PERIOD.EXECUTE && // Executable period
+            dispute.state === disputeConstants.STATE.OPEN // Open dispute
+        canExecute = dispute.state === disputeConstants.STATE.EXECUTABLE // Executable state
       }
 
+      // Wait for parallel requests to complete
+      ;[ruling, canRule] = await Promise.all(promises)
```

Comment:
those leading `;` are so gross. I think I would prefer it like `};`. Is it a linting thing?

Reply:
Also lets declare `promises` right before this. Makes it easier to figure out what promises are

Reply:
It's not a linting thing. It's a parser thing when not using EOL semicolons. See (http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding)[http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding].

They need to be declared before the `if (lastAppeal)` block, so they can be fired in parallel. I can change the var name to something less generic, but not the order of things.